recipes/cmake/3.x.x/conanfile.py

```diff
@@ -55,85 +55,89 @@ def validate(self):
-                                      self.install_folder.replace("\\", "/")))
             if self.settings.os == "Linux":
-                tools.replace_in_file(os.path.join(self._source_subfolder, "Utilities", "cmcurl", "CMakeLists.txt"),
+                replace_in_file(self, os.path.join(self.source_folder, "Utilities", "cmcurl", "CMakeLists.txt"),
                                       "list(APPEND CURL_LIBS ${OPENSSL_LIBRARIES})",
                                       "list(APPEND CURL_LIBS ${OPENSSL_LIBRARIES} ${CMAKE_DL_LIBS} pthread)")
-
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            cmake.configure()
             cmake.build()
 
     def package(self):
-        self.copy("Copyright.txt", dst="licenses", src=self._source_subfolder)
+        copy(self, "Copyright.txt", self.source_folder, os.path.join(self.package_folder, "licenses"), keep_path=False)
         if self.options.bootstrap:
-            with tools.chdir(self._source_subfolder):
-                autotools = AutoToolsBuildEnvironment(self)
+            with chdir(self, self.source_folder):
+                autotools = Autotools(self)
                 autotools.install()
         else:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            cmake.configure()
             cmake.install()
         rmdir(self, os.path.join(self.package_folder, "doc"))
 
     def package_id(self):
         del self.info.settings.compiler
 
     def package_info(self):
-        module_version = "{}.{}".format(Version(self.version).major, Version(self.version).minor)
-
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bindir))
-        self.env_info.PATH.append(bindir)
-
         self.buildenv_info.prepend_path("CMAKE_ROOT", self.package_folder)
-        self.env_info.CMAKE_ROOT = self.package_folder
+        self.runenv_info.define_path("CMAKE_ROOT", self.package_folder)
+        module_version = "{}.{}".format(Version(self.version).major, Version(self.version).minor)
         mod_path = os.path.join(self.package_folder, "share", f"cmake-{module_version}", "Modules")
         self.buildenv_info.prepend_path("CMAKE_MODULE_PATH", mod_path)
-        self.env_info.CMAKE_MODULE_PATH = mod_path
+        self.runenv_info.define_path("CMAKE_MODULE_PATH", mod_path)
         if not os.path.exists(mod_path):
             raise ConanException("Module path not found: %s" % mod_path)
 
         self.cpp_info.includedirs = []
+
+        # TODO remove once conan v2 is the only support and recipes have been migrated
+        if Version(conan_version).major < 2:
+            bindir = os.path.join(self.package_folder, "bin")
+            self.output.info("Appending PATH environment variable: {}".format(bindir))
+            self.runenv_info.append_path("PATH", bindir)
```

Comment:
Yes, but till Beta 5 is available, this will not run on Conan 2.0, which is what I am using for my development work per Diego's recommendations for new projects.

Reply:
Hi @uilianries,

I understand that but it would be nice if it were done in a manner that didn't break existing workflows in such a way that there is no fix until the next Beta release. What is wrong with leaving the conditional code until Beta 5 is available? Thanks

Reply:
Unfortunately, there is no way. There are more other points which are not compatible (e.g check_min_vs, is_apple_os, self.info.settings, ...) with Conan v1 and we need to deal. As it's still under Beta, we are fixing more and more things. We don't want to add more complexity with version conditions, that's why they are being removed from recipes.

Reply:
I have removed the conditional logic per your suggestion.